### PR TITLE
fix: Security group for secondary interface

### PR DESCRIPTION
**Pull request by jor2**
### Description

Previously when security groups for secondary interface was left empty, it would default to the default security group being attached, but on reapply terraform would find changes in the secondary vsi configs due to the default sg being attached, but the security group parameter empty in terraform.

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -135,7 +135,13 @@ resource "ibm_is_instance" "vsi" {
     }
     content {
       subnet = network_interfaces.value.id
-      security_groups = flatten([
+      security_groups = length(flatten([
```
Comment (jor2) [inline main.tf:138]: @ocofaigh any ideas how I could avoid code duplication here?

Comment (ocofaigh) [reply main.tf:138]: @jor2 Its pretty hard to follow that logic - could you maybe add some code comments to explain what its doing?

Comment (jor2) [reply main.tf:140]: Added some comments. It just checks if security group list is empty don't pass empty list, instead pass default security group. This avoid reapply failures.

Comment (ocofaigh) [reply main.tf:140]: ok thanks, comment is good. As for code duplication, have you tried using locals?
